### PR TITLE
Test DocumentCollection.php

### PR DESCRIPTION
**Pull request by AdrienHq**
|Q              |A  |
|---------------|---|
|Bug fix?       |N|
|New feature?   |N|	
|BC breaks?     |N|	
|Deprecations?  |N|	
|Fixed tickets  |N|	

Test for DocumentCollection.php

### CODE REVIEW

**Comment by dameert** [comment]
Hello, your branch "test/Documentcollection" is not up to date with our "develop" branch. Can you merge the latest version of develop in your branch?

**Comment by dameert** [comment]
It also seems than one of our automated tests are not passing (phpstan or phpcs) could you verify and resolve that also?

File: tests/Unit/Elasticsearch/Document/DocumentCollectionTest.php [hunk 1]
```diff
@@ -5,33 +5,33 @@
...
     {
-        $this->mockResponse = $this->createMock(ResponseInterface::class);
-        $this->mockResponse->method('getDocuments')
+        $mockResponse = $this->createMock(ResponseInterface::class);
+        $mockResponse->method('getDocuments')
             ->willReturn([
                 $this->createMock(DocumentInterface::class),
                 $this->createMock(DocumentInterface::class)
             ]);
+        $this->mockR = $mockResponse;
```
Comment (dameert) [inline tests/Unit/Elasticsearch/Document/DocumentCollectionTest.php:23]:
```suggestion
        $this->mockResponse = $mockResponse;
```

Comment (dameert) [reply tests/Unit/Elasticsearch/Document/DocumentCollectionTest.php:23]: Why did you change this to mockR, that is not a meaningfull name.

File: tests/Unit/Elasticsearch/Document/DocumentCollectionTest.php [hunk 2]
```diff
@@ -5,33 +5,33 @@
...
         $this->assertEquals(2, $collection->count());
     }
 
-    public function testGetIterator()
+    public function testGetIterator(): void
     {
-        $collection = DocumentCollection::fromResponse($this->mockResponse);
-        $this->assertEquals(2, \count($collection->getIterator()));
+        $collection = DocumentCollection::fromResponse($this->mockR);
+        $this->assertEquals(2, \iterator_count(($collection->getIterator())));
```
Comment (dameert) [inline tests/Unit/Elasticsearch/Document/DocumentCollectionTest.php:35]: I think you have to many ( ) here!
```suggestion
        $this->assertEquals(2, \iterator_count($collection->getIterator()));
```

File: tests/Unit/Elasticsearch/Document/DocumentCollectionTest.php [hunk 3]
```diff
@@ -5,33 +5,33 @@
...
 use EMS\CommonBundle\Elasticsearch\Response\ResponseInterface;
 use EMS\CommonBundle\Elasticsearch\Document\DocumentInterface;
-
 use PHPUnit\Framework\TestCase;
 
 class DocumentCollectionTest extends TestCase
 {
-    /** @var \ResponseInterface&\PHPUnit\Framework\MockObject\MockObject $mockResponse */
-    private $mockResponse;
+    /** @var letypehint $mockR */
```
Comment (dameert) [inline tests/Unit/Elasticsearch/Document/DocumentCollectionTest.php:12]: what is "letypehint"? should this not be ResponseInterface|MockObject?
```suggestion
    /** @var ResponseInterface|PHPUnit\Framework\MockObject\MockObject $mockResponse */
```

**Comment by dameert** [comment]
Looking at the build error, you should upgrade phpstan to ^7:
"phpunit/phpunit": "^7.0",

**Comment by dameert** [comment]
> Looking at the build error, you should upgrade phpstan to ^7:
> "phpunit/phpunit": "^7.0",

You can do this in the composer.json file, and then run "composer update phpunit/phpunit" to test it locally

File: composer.json [hunk 4]
```diff
@@ -45,7 +45,9 @@
 		"phpunit/phpunit" : "^5.7",
 		"squizlabs/php_codesniffer": "^3.4",
 		"symfony/phpunit-bridge": "^4.4",
-		"symfony/web-profiler-bundle": "^4.4"
+		"symfony/web-profiler-bundle": "^4.4",
+		"phpstan/extension-installer": "^1.0",
```
Comment (Davidmattei) [inline composer.json:49]: Mmmm what is this? Do we really need this, if so explain it to me :)

Comment (dameert) [reply composer.json:49]: I agree, the instructions also have a "manual" installation option, which is just adding 1 line of code to the phpstan configuration. The manual option is preferred

**Comment by dameert** [comment]
In fact this point was blocking because of a dependency issue. PHPunit needs to be at version 7 (I upgraded to 8.5) for this to work. I will close this PR in favor of #92

Both of your PR's have shown a bug in our code! Those are solved now! 👍

**Pull request closed without merging**